src/sema/statements.rs

```diff
@@ -878,45 +877,293 @@ fn statement(
             res.push(Statement::Assembly(resolved_asm.0, resolved_asm.1));
             Ok(resolved_asm.1)
         }
-        pt::Statement::Revert(loc, error, args) => {
-            if let Some(error) = error {
-                ns.diagnostics.push(Diagnostic::error(
-                    error.loc,
-                    format!("revert with custom error '{error}' not supported yet"),
-                ));
-                return Err(());
+        pt::Statement::Revert(loc, path, args) => {
+            if let Ok(stmt) = revert_pos_arg(loc, path, args, context, symtable, diagnostics, ns) {
+                res.push(stmt);
             }
 
-            let id = pt::Identifier {
-                loc: pt::Loc::File(loc.file_no(), loc.start(), loc.start() + 6),
-                name: "revert".to_string(),
+            Ok(false)
+        }
+        pt::Statement::RevertNamedArgs(loc, path, args) => {
+            if let Ok(stmt) = revert_named_arg(loc, path, args, context, symtable, diagnostics, ns)
+            {
+                res.push(stmt);
+            }
+            Ok(false)
+        }
+        pt::Statement::Error(_) => unimplemented!(),
+    }
+}
+
+fn revert_pos_arg(
```

Comment:
Here too, a small comment here would be nice

Reply:
I've added a comment